### DISCUSSION TITLE
Use Prompt while Progress/Live is running

### DISCUSSION THREAD

**Discussion by mon** [Textualize/rich#1164] [Q&A] [answered]
I cannot work out how to ask a question during a long-running task - I've tried making `visible=False` for all my tasks, manually refreshing, creating new `Console`s and similar, but the formatting always tends to collide. Here's a simple example of what happens:

```python
from rich.prompt import Confirm
from rich.progress import track

for s in track(['hello', 'world']):
    if not Confirm().ask('Continue?', default=True):
        break
```

Prints something like:
```
[2KContinue? [1;35m[0m [1;36m(y)[0m: Working... [38;5;237m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[0m [35m  0%[0m [36m-:--:--[0m
Working... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% -:--:--
```

The escape codes aren't a bad paste - they appear visibly in the terminal.

**Comment by qTipTip** [comment]
Hey! First of all, thanks for a great tool. `rich` is simply amazing!

I am also very much interested in this!

I've been furiously googling to find a way of doing this, to no avail. 
The need for this is certainly there, and I believe this would drive additional usage of `rich`.

**Comment by willmcgugan** [comment, marked as answer]
I'm afraid that Confirm and track will not work together. Confirm calls the system input() which assumes it has complete control over the terminal.

I am working on more dynamic features for Rich, so you may be able to do something like this in the future.

**Reply by infoseckev** [reply]
Does this mean we cannot get any input with Prompt or input() using Live? I have a layout updating with Live and would like user input. Prompt seems to freeze my layout until an input is given. Is there any way to get user input inside a Layout with Live?

**Reply by willmcgugan** [reply]
Not with the builtin `input`. Textual will have this capability eventually.
